julee_example/worker.py

```diff
@@ -0,0 +1,219 @@
+    # Setup logging first
+    setup_logging()
+
+    # Connect to Temporal server using environment variable
+    temporal_endpoint = os.environ.get("TEMPORAL_ENDPOINT", "localhost:7234")
+    logger.info(
+        "Starting julee_example Temporal worker",
+        extra={"temporal_endpoint": temporal_endpoint},
+    )
+
+    client = await get_temporal_client_with_retries(temporal_endpoint)
+
+    # Get Minio endpoint and create client for repositories
+    logger.debug("Preparing repository configurations")
+    minio_endpoint = os.environ.get("MINIO_ENDPOINT", "localhost:9000")
+
+    # Create Minio client for repositories
+    # minio.Minio implements the MinioClient protocol
+    minio_client: MinioClient = Minio(  # type: ignore[assignment]
+        endpoint=minio_endpoint,
+        access_key="minioadmin",
+        secret_key="minioadmin",
+        secure=False,
+    )
+
+    # Instantiate temporal repository classes for activity registration
+    logger.debug("Creating Temporal Activity repository implementations")
+    temporal_assembly_repo = TemporalMinioAssemblyRepository(
+        client=minio_client
+    )
+    temporal_assembly_spec_repo = (
+        TemporalMinioAssemblySpecificationRepository(client=minio_client)
+    )
+    temporal_document_repo = TemporalMinioDocumentRepository(
+        client=minio_client
+    )
+    temporal_knowledge_config_repo = (
+        TemporalMinioKnowledgeServiceConfigRepository(client=minio_client)
+    )
+    temporal_knowledge_query_repo = (
+        TemporalMinioKnowledgeServiceQueryRepository(client=minio_client)
+    )
+
+    # Create temporal knowledge service for activity registration
+    temporal_knowledge_service = TemporalKnowledgeService()
+
+    # Collect all activities from repository instances
+    # The @temporal_activity_registration decorator automatically wraps
+    # all async methods as Temporal activities
+    activities = [
+        # Assembly repository activities
+        temporal_assembly_repo.generate_id,
+        temporal_assembly_repo.save,
+        temporal_assembly_repo.get,
+        # Assembly specification repository activities
+        temporal_assembly_spec_repo.generate_id,
+        temporal_assembly_spec_repo.save,
+        temporal_assembly_spec_repo.get,
+        # Document repository activities
+        temporal_document_repo.generate_id,
+        temporal_document_repo.save,
+        temporal_document_repo.get,
+        # Knowledge service config repository activities
+        temporal_knowledge_config_repo.generate_id,
+        temporal_knowledge_config_repo.save,
+        temporal_knowledge_config_repo.get,
+        # Knowledge service query repository activities
+        temporal_knowledge_query_repo.generate_id,
+        temporal_knowledge_query_repo.save,
+        temporal_knowledge_query_repo.get,
+        # Knowledge service activities
+        temporal_knowledge_service.register_file,
+        temporal_knowledge_service.execute_query,
+    ]
```

Comment:
Given the automated decorator that we're using, I think we should be able to remove this boiler-plate too. Adding a TODO